httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java

```diff
@@ -272,6 +274,20 @@ public SSLContext build() throws NoSuchAlgorithmException, KeyManagementExceptio
         return sslcontext;
     }
 
+    public SSLContext buildWithProvider(final Provider provider) throws NoSuchAlgorithmException, KeyManagementException {
```

Comment:
Fair point. However I am not sure this change should go into 4.4. In 5.0 we can change APIs as we deem fit.